src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -139,41 +137,35 @@ private Future<Void> processInventory(CompositePoLine compPOL, boolean deleteHol
-        if (logger.isDebugEnabled()) {
-          String deletedIds = pieces.stream().map(Piece::getId).collect(Collectors.joining(","));
-          logger.debug(String.format("Pieces were removed : %s", deletedIds));
-        }
-      })
-      .mapEmpty();
+          if (logger.isDebugEnabled()) {
+            String deletedIds = pieces.stream().map(Piece::getId).collect(Collectors.joining(","));
+            logger.debug(String.format("Pieces were removed : %s", deletedIds));
+          }
+        })
+        .mapEmpty();
     }
     if (PoLineCommonUtil.isItemsUpdateRequired(compPOL)) {
-        if (compPOL.getCheckinItems()) { // independent workflow
-          return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
-        } else { // synchronized workflow
-          return (deleteHoldings) ? processInventoryHoldingWithItems(compPOL, requestContext) : processInventoryOnlyWithItems(compPOL, requestContext);
-        }
+      if (compPOL.getCheckinItems()) { // independent workflow
+        return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
+      } else { // synchronized workflow
+        return (deleteHoldings) ? processInventoryHoldingWithItems(compPOL, requestContext) : processInventoryOnlyWithItems(compPOL, requestContext);
+      }
     }
     if (PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)) {
-        if (compPOL.getCheckinItems()) { // independent workflow
-          return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
-        } else { // synchronized workflow
-          return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) :
-            deleteExpectedPieces(compPOL, requestContext).mapEmpty();
-        }
+      if (compPOL.getCheckinItems()) { // independent workflow
+        return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
+      } else { // synchronized workflow
+        return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) :
+          deleteExpectedPieces(compPOL, requestContext).mapEmpty();
+      }
     }
 
     return Future.succeededFuture();
   }
 
   private Future<Void> processInventoryOnlyWithHolding(CompositePoLine compPOL, RequestContext requestContext) {
     return deleteExpectedPieces(compPOL, requestContext)
-      .compose(deletedPieces -> {
-        List<String> holdingIds = compPOL.getLocations().stream()
-          .map(Location::getHoldingId)
-          .filter(Objects::nonNull)
-          .collect(toList());
-        return inventoryHoldingManager.getHoldingsByIds(holdingIds, requestContext);
-      })
+      .compose(deletedPieces -> inventoryHoldingManager.getHoldingsForAllLocationTenants(compPOL, requestContext))
       .compose(holdings -> deleteHoldings(holdings, requestContext))
```

Comment:
if holdings are located in different tenants - this call deleteHoldings() will not work, because it deletes only from single tenant. If it is hard to implement now, we can extract this into new separate story

Reply:
Created https://folio-org.atlassian.net/browse/MODORDERS-1103